docs/docs/config/kscan.md

```diff
@@ -98,6 +94,25 @@ Assuming the switches connect each GPIO pin to the ground, the [GPIO flags](http
     };
 ```
 
+By default, a switch will drain current through the internal pull up/down resistor whenever it is pressed. This is not ideal for a toggle switch, where the switch may be left in the "pressed" state for a long time. Enabling `toggle-mode` will make the driver flip between pull up and down as the switch is toggled to optimize for power. For `toggle-mode` to work correctly each pole of the switch needs a dedicated GPIO pin.
```

Comment:
Don't really like the wording of this (you copied this directly, just noticing it now):

> Enabling `toggle-mode` will make the driver flip between pull up and down as the switch is toggled to optimize for power

Maybe "will make the driver enable/disable the internal pull up/down as needed when the switch is toggled to minimize power draw"?